File: at_coder/contest_043/C/a.py
import sys
import math

def main():
    sys.stdin = open('input.txt', 'r')

    input()
    a = list(map(int, input().split()))

    y = round(sum(a)/len(a))

    ans = 0
    for i in a:
        ans += pow((i - y), 2)

    print(ans)


if __name__ == '__main__':
    main()

File: at_coder/contest_043/C/a.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    
    freopen("input.txt", "r", stdin);

    int N;

    cin>>N;

    vector<int> dollars;

    for(int i=0; i<N; i++) {
        int temp;
        cin>>temp;
        dollars.push_back(temp);
    }

    float y = 0.0; 
    int ans = 0;
    y = accumulate(dollars.begin(), dollars.end(), y) / dollars.size();
    y = round(y);

    for(int x: dollars) {
        ans += pow(x - y, 2);

    }

    cout<<ans<<'\n';

    return 0;
}